File: bridges/voiceover/Sources/ScreenReaderWire/Commands/Echo.swift
// ROLE: entity, `echo`'s params and result.

public struct EchoParams: Codable, Equatable, Sendable {
	public var payload: JSONValue

	public init(payload: JSONValue) {
		self.payload = payload
	}
}

public struct EchoResult: Codable, Equatable, Sendable {
	public var payload: JSONValue

	public init(payload: JSONValue) {
		self.payload = payload
	}
}

File: bridges/nvda/addon/globalPlugins/nvdaMcpBridge/domain/controllers/commands/echo.py
# nvdaMcpBridge domain -- EchoHandler: diagnostic payload round-trip.
# Copyright (C) 2026 Marlon Brandao de Sousa. GPL-2. See COPYING.txt.
# ROLE: command handler for `echo`.

from __future__ import annotations

from typing import TYPE_CHECKING, Any

from .... import protocol
from .command_handler import CommandHandler

if TYPE_CHECKING:
	from .session_context import SessionContext


class EchoHandler(CommandHandler):
	def execute(self, ctx: SessionContext, request: protocol.Request) -> Any:
		params = protocol.from_dict(protocol.EchoParams, request.params)
		return protocol.EchoResult(payload=params.payload)
